docs/src/test/scala/sample/scaladsl/TransactionsExample.scala

```diff
@@ -0,0 +1,59 @@
+package sample.scaladsl
+
+import akka.kafka.scaladsl.Consumer.Control
+import akka.kafka.scaladsl.{Consumer, Producer}
+import akka.kafka.{ProducerMessage, Subscriptions}
+import akka.stream.scaladsl.{RestartSource, Sink}
+import org.apache.kafka.clients.producer.ProducerRecord
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class TransactionsSink extends ConsumerExample {
+  def main(args: Array[String]): Unit = {
+    // #transactionalSink
+    val control =
+      Consumer.transactionalSource(consumerSettings, Subscriptions.topics("source-topic"))
+        .via(business)
+        .map { msg =>
+          ProducerMessage.Message(
+            new ProducerRecord[Array[Byte], String]("sink-topic", msg.record.value), msg.partitionOffset)
+        }
+        .runWith(Producer.transactionalSink(producerSettings, "transactional-id"))
+    // #transactionalSink
+
+    terminateWhenDone(control)
+  }
+}
+
+class TransactionsFailureRetryExample extends ConsumerExample {
+  def main(args: Array[String]): Unit = {
+    // #transactionalFailureRetry
+    var innerControl = null.asInstanceOf[Control]
```

Comment:
consider 
```
var innerControl : Control = _
```  
to avoid the noisy `asInstanceOf`

Reply:
That's actually not possible (results in a compiler error) because `innerControl` is a variable.  See this SO answer for a summary as to why: https://stackoverflow.com/a/36955530/895309

Reply:
Interesting. I didn't know about the scope difference. Thanks for the link.
How about:
```
var innerControl = null.asInstanceOf[Control]
// vs
var innerControl:Control = null
```
?

Reply:
Yes, that works.